### PR TITLE
Structured Data 2021 Markdown

### LINKED ISSUE #2174: Structured Data 2021
# Part I Chapter 4: Structured Data

If you're interested in contributing to the Structured Data chapter of the 2021 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @jonoalderson | @jonoalderson @cyberandy | @kevinmarks @vdwijngaert @jvandriel @philbarker | @GregBrimble  | @jvandriel @JasmineDWillson | @rviscomi |

<details>
<summary><b>Expand for more information about each role</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 31**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **June 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 30**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **July 1 - 31**: HTTP Archive runs the July crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **September 30**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [ ] **October 31**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **November 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **December 1**: Target launch date 🚀

## Chapter resources

Refer to these 2021 Structured Data resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata

[~google-doc]: https://docs.google.com/document/d/19KDSv4olAXUHUV6hq4X4Cb-lNziqvVesgXXxVktrw4c/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2021/structured-data
[~google-sheets]: https://docs.google.com/spreadsheets/d/1uEA217YjpX1xdRVBnIz9ekjpjIrpGIh5GwUxvnzJNig/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2021/structured-data.md

### PR DESCRIPTION
**Pull request by vdwijngaert**
Closes #2174

### CODE REVIEW

**Comment by tunetheweb** [comment]
OK this is staged here: https://20211111t105151-dot-webalmanac.uk.r.appspot.com/en/2021/structured-data

@jonoalderson, @cyberandy, @vdwijngaert, @philbarker, @GregBrimble, @jvandriel, @JasmineDWillson can you review please?

**Review by jonoalderson — APPROVED** [review]
Nice nice grammar good.

**Comment by GregBrimble** [comment]
Looks like we're inadvertently linking to a GitHub profile in the JSON-LD relationships table for `@graph`.

**Comment by tunetheweb** [comment]
> Looks like we're inadvertently linking to a GitHub profile in the JSON-LD relationships table for `@graph`.

Good spot! Fixed.

File: src/content/en/2021/structured-data.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+description: Structured Data chapter of the 2021 Web Almanac covering adoption and impact of schema.org, RDFa, Microdata and more.
+authors: [jonoalderson, cyberandy]
+reviewers: [vdwijngaert, philbarker]
+analysts: [GregBrimble]
+translators: [tunetheweb]
+editors: [jvandriel, JasmineDWillson]
+results: https://docs.google.com/spreadsheets/d/1uEA217YjpX1xdRVBnIz9ekjpjIrpGIh5GwUxvnzJNig/edit?usp=sharing
+jonoalderson_bio: Jono Alderson is a digital strategist, marketing technologist, and full stack developer. He enjoys dabbling with website performance, technical SEO, schema.org and all things structured data.
+cyberandy_bio: Andrea Volpini is the CEO of WordLift, and is currently focusing on the semantic web, SEO and artificial intelligence.
+featured_quote: Describing our content using structured data enables machines to treat webpages and websites as databases. That creates a wealth of possibilities for business, technology, and society.
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:13]: Please fill in featured stats below this.

File: src/content/en/2021/structured-data.md [hunk 2]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+  )
+}}
+
+When we evaluate the types of RDFa, we can see that the `foaf:image` syntax is present on far more pages than any other type—on upwards of 0.86% of all pages in our data set. Whilst that may seem like a small proportion, it represents over ~65,000 pages, and over 60% of the total RDFa markup that we discovered.
+
+Beyond this outlier, the use of RDFa diminishes and fragments considerably, though there are still some interesting discoveries to explore.
+
+##### On FOAF
+
+[FOAF](http://xmlns.com/foaf/spec/) is a linked data dictionary of people-related terms, created in the early-2000s. It can be used to describing _people_, _groups_ and _documents_.
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:165]: Does FOAF stand for something? Should we mention that?

File: src/content/en/2021/structured-data.md [hunk 3]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+FOAF uses W3C's RDF syntax and in its <a hreflang="en" href="https://web.archive.org/web/20140331104046/http://www.foaf-project.org/original-intro">original introduction</a> was explained as follows:
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:167]: RDF or RDFa?

```suggestion
FOAF uses W3C's RDFa syntax and in its <a hreflang="en" href="https://web.archive.org/web/20140331104046/http://www.foaf-project.org/original-intro">original introduction</a> was explained as follows:
```

File: src/content/en/2021/structured-data.md [hunk 4]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+<p class="note">"Consider a Web of inter-related home pages, each describing things of interest to a group of friends. Each new home page that appears on the Web tells the world something new, providing factoids and gossip that make the Web a mine of disconnected snippets of information. FOAF provides a way to make sense of all this."</p>
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:169]: What is this a quote from?

File: src/content/en/2021/structured-data.md [hunk 5]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+Anecdotally, we can attribute a prominence of `foaf` markup in our results to sites running on older versions of the Drupal CMS, which historically added `typeof="foaf:image"` and `foaf:document` markup to its HTML by default.
+
+##### On other notable RDFa findings
+
+As well as FOAF properties, various other standards and syntaxes show up in our list.
+
+Notably, we can see several `sioc` properties, such as `sioc:item` (0.24% of pages) and `sioc:useraccount` (0.03% of pages). <a hreflang="en" href="https://www.w3.org/Submission/sioc-spec/">SIOC</a> is a standard designed to describe structured data relating to _online communities_, such as message boards, forums, wikis and blogs.
+
+We can also see a <a hreflang="en" href="https://www.w3.org/TR/skos-primer/">SKOS</a> property—`skos:concept`—on 0.04% of pages. SKOS is another standard, which aims to provide a way of describing taxonomies and classifications (e.g., tags, data sets, and so on).
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:179]: Is SKOS an acronym for something?

File: src/content/en/2021/structured-data.md [hunk 6]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+#### Dublin Core
+
+<a hreflang="en" href="https://dublincore.org/">Dublin Core</a> is a vocabulary interoperable with linked data standards that was originally conceived in Dublin, Ohio in 1995 at an OCLC (Online Computer Library Center) and NCSA (National Center for Supercomputing Applications) workshop.
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:183]: Should we add links to OCLC and NCSA?

```suggestion
<a hreflang="en" href="https://dublincore.org/">Dublin Core</a> is a vocabulary interoperable with linked data standards that was originally conceived in Dublin, Ohio in 1995 at an 'Online Computer Library Center](https://www.oclc.org/en/home.html) (OCLC) and <a hreflang="en" href="https://www.ncsa.illinois.edu/">National Center for Supercomputing Applications</a> (NCSA) workshop.
```

File: src/content/en/2021/structured-data.md [hunk 7]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+* [Microformats](http://microformats.org/) (and <a hreflang="en" href="https://microformats.org/wiki/microformats2">microformats2</a>)
+* [RDFa](https://en.wikipedia.org/wiki/RDFa), [Microdata](https://en.wikipedia.org/wiki/Microdata_(HTML)) and <a hreflang="en" href="https://json-ld.org/">JSON-LD</a>
+
+Collectively, these provide a broad overview of different use-cases and scenarios; and include both legacy standards and modern approaches (e.g., microformats vs JSON-LD).
+
+Before we explore specific usage across the various structured data types, we should briefly explore some caveats.
+
+### Data caveats
+
+#### 1. The influence of Content Management Systems
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:94]: Headings should have at least one sentence and not be empty (so called "Stacked headings", which are used to group headings but don't have any content themselves). "A heading is like a chapter title; it must have at least a sentence of information below it. Stacked headings can indicate inefficient organization of information."

File: src/content/en/2021/structured-data.md [hunk 8]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+That the most popular attribute type is `dc:title` (on 0.70% of pages) comes as no surprise; but it is interesting to see that `dc:language` is next (above common descriptors like _description_, _subject_ and _publisher_) with a penetration of 0.49%. This makes sense, when you consider that Dublin Core is often used in multilingual metadata management systems.
+
+It's also interesting to see the relatively prominent appearance of `dc:relation` (on 0.16% of pages)—an attribute that is capable of expressing relationships between different concepts.
+
+While it might seem to many that Schema.org is predominant in the context of SEO, the role of DC remains pivotal because of its broad interpretation of concepts and its deep roots in the _linked open data movement_.
+
+#### Social metadata
+
+##### Open Graph
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:209]: Headings should have at least one sentence and not be empty (so called "Stacked headings", which are used to group headings but don't have any content themselves). "A heading is like a chapter title; it must have at least a sentence of information below it. Stacked headings can indicate inefficient organization of information."

File: src/content/en/2021/structured-data.md [hunk 9]
```diff
@@ -17,4 +19,568 @@ featured_stat_3: TODO
...
+
+A web made of deeply connected, structured data which powers a more integrated world has long been a science-fiction dream. But perhaps, not for much longer. As these standards continue to evolve, and their adoption continues to grow, we pave a road towards an exciting future.
+
+### Future years
+
+As this is the first time this chapter has been included in the Web Almanac, we don't have any data we can use to explore change over time.
+
+In future years, we'll be able to assess how the distribution, types, and formats of structured data have changed across the web. This may provide valuable insight into growing/declining technologies, strategies, and opportunities.
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:584]: Maybe rephrase the first sentence to be more positive?

Something along the lines of "In future years we hope to be able to continue this analysis the analysis started here and map the evolution of structured data usage over time." which sounds more like this is an opportunity, than an excuse or something we've missed or an unable to cover in this chapter?

**Review by tunetheweb — COMMENTED** [review]
Great job all!

@jonoalderson @cyberandy I've commit the basic copyedits to the chapter here: https://github.com/HTTPArchive/almanac.httparchive.org/pull/2466/commits/cd2eb7b224287e5adf2af0526e801504ff5ef94e. Mostly it was to tone down overuse of quotes of emphasis that I felt was unnecessary.

I've made comments below on some other changes for you to consider.

File: src/content/en/2021/structured-data.md (hunk 1 above)
Comment (jonoalderson) [reply src/content/en/2021/structured-data.md:13]: Done!

**Comment by tunetheweb** [comment]
Here's what the featured quote looks like on the home page:

image

**Comment by jonoalderson** [comment]
Think I'll tweak those featured stats for a bit more pizazz. Added to tomorrow's to-do list.

File: src/content/en/2021/structured-data.md [hunk 10]
```diff
@@ -1,20 +1,595 @@
...
+
+Of all of the Facebook tags that we detected, there are only three tags with significant adoption.
+
+Those are `fb:app_id`, `fb:admins`, and `fb:pages`; which we found on 6.06%, 2.63% and 0.86% of pages respectively.
+
+These tags are used to explicitly relate a webpage to a Facebook Page/Brand, or to grant permissions to a user (or users)  who administrates those profiles.
+
+Anecdotally, it's unclear how well these are supported by Facebook. The platform has gone through radical changes over the past few years, and their technical documentation hasn't been well-maintained. However, many content management systems, templates and best practice guides—as well as some of Facebook's debugging tools—still include and make reference to them.
+
+##### Microformats (and microformats2)
```
Comment (tunetheweb) [inline src/content/en/2021/structured-data.md:302]: Might wanna consider dropping the brackets:

```suggestion
##### Microformats and microformats2
```

As looks a bit odd in the Table of Contents at the moment:


image

**Comment by tunetheweb** [comment]
I've added a "Big number callout": https://20211111t105151-dot-webalmanac.uk.r.appspot.com/en/2021/structured-data#relationship-depth

image

As well as breaking up the text a bit, it also allows us to link the data and query for those wanting to dive into it more:

image

File: src/content/en/2021/structured-data.md (hunk 2 above)
Comment (jonoalderson) [reply src/content/en/2021/structured-data.md:170]: Clarified.

File: src/content/en/2021/structured-data.md (hunk 4 above)
Comment (jonoalderson) [reply src/content/en/2021/structured-data.md]: Clarified

File: src/content/en/2021/structured-data.md (hunk 5 above)
Comment (jonoalderson) [reply src/content/en/2021/structured-data.md:186]: Clarified

File: src/content/en/2021/structured-data.md (hunk 9 above)
Comment (jonoalderson) [reply src/content/en/2021/structured-data.md]: Done

**Comment by rviscomi** [comment]
Happy to merge this. Can follow up with any small changes in a separate PR as they come up. Great work everyone! 🚀
